src/pal/src/misc/sysinfo.cpp

```diff
@@ -308,8 +312,25 @@ DWORD
 PALAPI
 GetCurrentProcessorNumber()
 {
-    // TODO: implement this
-    return 0;
+#if HAVE_SCHED_GETCPU
+    return sched_getcpu();
+#endif
+
+    return -1;
+}
+
+BOOL
+PALAPI
+PAL_HasFunctionality(
```

Comment:
Do we really need this function? I guess that checking if the GetCurrentProcessorNumber returned -1 or not should be good enough.